### PR TITLE
EASY-2770: (part 2) urn:nbn is missing from dataset.xml

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-2770: (part 2) urn:nbn is missing from dataset.xml

#### When applied it will...
* 
* 
* 

#### Related pull requests on github

repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala [hunk 1]
```diff
@@ -302,8 +302,8 @@ object DDM extends DebugEnhancedLogging {
   }
 
   private def isDdmId(bi: BasicIdentifier): Boolean = {
-    // these ID's are generated by easy-ingest-flow/EASY-I, not converted from DDM
-    !Seq("PID", "DMO_ID", "AIP_ID").contains(bi.getScheme)
+    // generated into EMD by easy-ingest-flow/EASY-I, not schema valid in DDM
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala:305]: do you mean to say that with these identifiers in dataset.xml the file is no longer valid?

Comment (jo-pol) [reply src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala:305]: When omitting either one from this list we would indeed get 
```
org.xml.sax.SAXParseException; lineNumber: 18; columnNumber: 44; cvc-elt.4.2: Cannot resolve 'id-type:PID' to a type definition for element 'dct:identifier'.
org.xml.sax.SAXParseException; lineNumber: 19; columnNumber: 47; cvc-elt.4.2: Cannot resolve 'id-type:DMO_ID' to a type definition for element 'dct:identifier'.
```
at   https://github.com/DANS-KNAW/easy-fedora2vault/blob/d7e91a52a1fb8bad3979bce6ab35a20afce0359c/src/test/scala/nl/knaw/dans/easy/fedora2vault/DdmSpec.scala#L109

Comment (jo-pol) [reply src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala]: Ah, `id-type:PID` should have been `id-type:URN`

**Review by jo-pol — COMMENTED** [review]
~~I don't understand. The tests succeed with intellij but fail with `mvn clean install`. When I rebuild blessed/master it now fails with the same errors. It succeeded before: https://travis-ci.org/github/DANS-KNAW/easy-fedora2vault/jobs/723028581~~

File: src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala (hunk 1 above)
Comment (lindareijnhoudt) [reply src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala]: do you know if the fedora-id is in the resulting dataset.xml?

Comment (jo-pol) [reply src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala]: Doesn't look like that https://github.com/DANS-KNAW/easy-fedora2vault/blob/016caf55f2dfa07800a1d5749e672b8d4540333a/src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala#L304-L307

File: src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala [hunk 2]
```diff
@@ -66,7 +66,7 @@ object DDM extends DebugEnhancedLogging {
        <ddm:accessRights>{ emd.getEmdRights.getAccessCategory }</ddm:accessRights>
      </ddm:profile>
      <ddm:dcmiMetadata>
-       { emd.getEmdIdentifier.getDcIdentifier.asScala.filter(isDdmId).map(bi => <dct:identifier xsi:type={ idType(bi) }>{ bi.getValue }</dct:identifier>) }
+       { emd.getEmdIdentifier.getDcIdentifier.asScala.filter(_.getScheme != "DMO_ID").map(bi => <dct:identifier xsi:type={ idType(bi) }>{ bi.getValue }</dct:identifier>) }
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala:69]: why the exclusion of DMO_ID? this is the fedora identifier. is it added elsewhere?

Comment (jo-pol) [reply src/main/scala/nl/knaw/dans/easy/fedora2vault/DDM.scala:69]: If remembered well it didn't validate somewhere and broke tests.

**Pull request merged**